.editorconfig

```diff
@@ -0,0 +1,388 @@
+
+# Indentation and spacing
+tab_width = 4
+indent_size = 4
+indent_style = space
+
+# New line preferences
+end_of_line = crlf
+insert_final_newline = false
+trim_trailing_whitespace = true
+
+#### Build files ####
+
+# Solution files
+[*.{sln,slnx}]
+tab_width = 4
+indent_size = 4
+indent_style = tab
+
+# Configuration files
+[*.{json,xml,yml,config,runsettings}]
+indent_size = 2
+
+# MSBuild files
+[*.{slnf,props,targets,projitems,csproj,shproj}]
+indent_size = 2
+
+#### Source files ####
+
+# Markdown files
+[*.md]
+indent_size = 2
+insert_final_newline = true
+
+# XAML files
+[*.xaml]
+indent_size = 4
+
+# C# files
+[*.cs]
+
+#### .NET Coding Conventions ####
+
+# this. and Me. preferences
+dotnet_style_qualification_for_event = false:warning
+dotnet_style_qualification_for_field = true:warning
+dotnet_style_qualification_for_method = false:warning
+dotnet_style_qualification_for_property = false:warning
+
+# Language keywords vs BCL types preferences
+dotnet_style_predefined_type_for_locals_parameters_members = true:warning
+dotnet_style_predefined_type_for_member_access = true:warning
+
+# Parentheses preferences
+dotnet_style_parentheses_in_arithmetic_binary_operators = always_for_clarity:silent
+dotnet_style_parentheses_in_other_binary_operators = always_for_clarity:silent
+dotnet_style_parentheses_in_other_operators = never_if_unnecessary:silent
+dotnet_style_parentheses_in_relational_binary_operators = always_for_clarity:silent
+
+# Modifier preferences (set to silent until https://github.com/dotnet/roslyn/issues/52904 is resolved)
+dotnet_style_require_accessibility_modifiers = for_non_interface_members:silent
+
+# Code block preferences
+dotnet_style_allow_multiple_blank_lines_experimental = false:warning
+dotnet_style_allow_statement_immediately_after_block_experimental = false:warning
+
+# Expression-level preferences
+csharp_style_deconstructed_variable_declaration = true:suggestion
+csharp_style_inlined_variable_declaration = true:silent
+csharp_style_throw_expression = true:suggestion
+dotnet_style_coalesce_expression = true:suggestion
+dotnet_style_collection_initializer = true:suggestion
+dotnet_style_explicit_tuple_names = true:suggestion
+dotnet_style_null_propagation = true:suggestion
+dotnet_style_object_initializer = true:suggestion
+dotnet_style_prefer_auto_properties = true:silent
+dotnet_style_prefer_conditional_expression_over_assignment = true:silent
+dotnet_style_prefer_conditional_expression_over_return = true:silent
+dotnet_style_prefer_inferred_anonymous_type_member_names = true:suggestion
+dotnet_style_prefer_inferred_tuple_names = true:suggestion
+dotnet_style_prefer_is_null_check_over_reference_equality_method = true:warning
+
+# Field preferences
+dotnet_style_readonly_field = true:warning
+
+#### C# Coding Conventions ####
+
+# var preferences
+csharp_style_var_elsewhere = false:warning
+csharp_style_var_for_built_in_types = false:warning
+csharp_style_var_when_type_is_apparent = false:warning
```

Comment:
Are you expecting these to fail CI but they are not?

This is by-design. The compiler doesn't know about the `option_key = option_value:severity` syntax.
You have to do `dotnet_diagnostic.IDExxxx.severity = warning`
`var` preferences are IDE0007 and IDE0008

Reply:
Resolving this as that MSBuild property to enable all IDE warnings did fix this 🙌